pkg/apiserver/resthandler.go

```diff
@@ -294,6 +295,22 @@ func UpdateResource(r RESTUpdater, ctxFn ContextFunc, namer ScopeNamer, codec ru
 			return
 		}
 
+		// Check to see if a resource version has been supplied.
+		// If there is no resource version then fail this update request.
+		// ALTERNATIVE: Update at the most recent resource version.
```

Comment:
We should not coerce to the most recent version by default. That would make it too easy to clobber the desired state.

Seems like this should be done in ValidateObjectMetaUpdate:
https://github.com/GoogleCloudPlatform/kubernetes/blob/9bc2b0c2dbb92e25a48c5aa21e3d3e2ef7d18687/pkg/api/validation/validation.go#L216

It can't be done here, because we allow creation via PUT as well, in which case there is no resourceVersion -- see line 327 below.